### ISSUE TITLE
transrate Debian packaging

### ISSUE THREAD

**Issue by mr-c** [Blahah/transrate#160]
Oh my, so many dependencies! Alas the biggest delay will be waiting for leaf packages to be accepted so those packages that depend on them can be sent in.

Dependencies that need packaging (default) or updates (as marked):
```
[ ] transrate
 + [ ] transrate-tools
 + [ ] salmon
 |  + [ ] libcereal-dev
 |  + [ ] libgff-dev
 |  + [ ] libspdlog-dev
 |  + [ ] libjellyfish-2.0-dev (update)
 |  + [ ] libbwa-dev (update)
 + [ ] snap
 + [ ] ruby-fix-trinity-output
 + [ ] ruby-trollop
 + [ ] ruby-yell
```

- [ ] Get membership to Debian Ruby team (my application has been submitted)
https://wiki.debian.org/Teams/Ruby/Packaging#Joining_the_pkg-ruby-extras_team

http://anonscm.debian.org/cgit/debian-med/cereal.git
- [ ] libcereal-dev: needs Intent To Package (ITP) bug filed
- [ ] libcereal-dev: fix `debian/copyright`
- [ ] libcereal-dev: drop extra license files from `usr/include/cereal/external/`
- [ ] libcereal-dev: sponsored uploade
- [ ] libcereal-dev: acceptance!

http://anonscm.debian.org/cgit/debian-med/libgff.git
- [ ] libgff-dev: ITP bug filed
- [ ] libgff-dev: needs description
- [ ] libgff-dev: needs watch file
- [ ] libgff-dev: sponsored upload
- [ ] libgff-dev: acceptance!

 http://anonscm.debian.org/cgit/debian-med/spdlog.git
- [ ] libspdlog: ITP bug
- [ ] libspdlog: better description
- [ ] libspdlog: sponsored upload
- [ ] libspdlog: acceptance!

 http://anonscm.debian.org/cgit/debian-med/jellyfish.git
- [ ] libjellyfish-2.0-dev: sponsored upload
- [ ] libjellyfish-2.0-dev: acceptance!

 http://anonscm.debian.org/cgit/debian-med/bwa.git
- [ ] libbwa-dev: sponsored upload
- [ ] libbwa-dev: acceptance!

http://anonscm.debian.org/cgit/debian-med/salmon.git
- [ ] salmon: ITP bug
- [ ] salmon: manpage
- [ ] salmon: track down missing source issue re: `doc/build/html/_static/jquery.js` & `underscore.js`
- [ ] salmon: tweak build to link dynamically to libraries.
- [ ] salmon: sponsored upload
- [ ] salmon: acceptance!

Ruby packages:
https://wiki.debian.org/Teams/Ruby/Packaging

- [ ] transrate: ITP bug
- [ ] transrate: gem2deb & create repo
- [ ] transrate: add patch to remove use of ruby-bindeps
- [ ] transrate: sponsored upload
- [ ] transrate: acceptance!

- [ ] ruby-fix-trinity-output: ITP bug
- [ ] ruby-fix-trinity-output: gem2deb & create repo
- [ ] ruby-fix-trinity-output: sponsored upload
- [ ] ruby-fix-trinity-output: acceptance!

- [ ] ruby-trollop: ITP bug
- [ ] ruby-trollop: gem2deb & create repo
- [ ] ruby-trollop: sponsored upload
- [ ] ruby-trollop: acceptance!

- [ ] ruby-yell: ITP bug
- [ ] ruby-yell: gem2deb & create repo
- [ ] ruby-yell: sponsored upload
- [ ] ruby-yell: acceptance!

**Comment by mr-c** [comment]
This will be a three-phase submission:

1) transrate-tools, snap-aligner, ruby-fix-trinity-output, ruby-trollop, ruby-yell, libcereal-dev, libgff-dev, libspdlog-dev, libjellyfish-2.0-dev, libbwa-dev (update)
2) salmon
3) transrate itself

**Comment by Blahah** [comment]
@mr-c this is heroic! And very much appreciated. What can we do to support it?

**Comment by mr-c** [comment]
@Blahah Thanks :-) The licenses need clarifying, see https://github.com/Blahah/transrate/issues/159.

In general it seems that the Ruby community is a fan of the MIT license and not putting this in each file nor having a copyright line. Alas this is rather confusing as
[multiple versions exist, all with the same name](https://en.wikipedia.org/wiki/MIT_License#Various_versions). When documenting the license for Debian I can use the short name 'expat' if it exactly matches [that variant](http://www.jclark.com/xml/copying.txt); otherwise it creates a delay.

Ideally the [program `licensecheck`](https://anonscm.debian.org/cgit/collab-maint/devscripts.git/tree/scripts/licensecheck.pl) (from the `devscripts` package) would give me the correct information when scanning source trees.

```
(sid)255 mcrusoe@localhost:~/debian/transrate-tools$ licensecheck `git ls-files`
src/bam-read.cpp: *No copyright* UNKNOWN
src/bam-read.h: *No copyright* UNKNOWN
src/pileup.cpp: *No copyright* UNKNOWN
src/pileup.h: *No copyright* UNKNOWN
src/segmenter.cpp: *No copyright* UNKNOWN
src/segmenter.h: *No copyright* UNKNOWN
```

**Comment by Blahah** [comment]
Can I deal with that by just putting a LICENSE file in the repo (see https://github.com/Blahah/transrate-tools/commit/215f661ac19fe847da62f49f56e6b3599aa81b72)?

**Comment by mr-c** [comment]
For transrate-tools & #159 yes, that is required; thanks!

A copyright line in each file would be nice too.

**Comment by Blahah** [comment]
You're right, I never put the license in every file because I find it aesthetically unpleasant - a distraction from the pure utility of the code. But now that you mention it, I see why it would be useful! 

Is there some convention for stating the license/copyright as simply as possible?

**Comment by mr-c** [comment]
:-)

I know of no convention, alas. Some licenses give specific guidance on how to use them in a source file; many do not.

For the MIT family of licenses it appears that the best bet is to include the entire license in the header as it is unversioned and so short. See: https://anonscm.debian.org/cgit/collab-maint/devscripts.git/tree/scripts/licensecheck.pl#n593

(DIB lab had the same issue with the BSD license; I ended up adding the full text of the license to every file during a copyright/license audit brought about because seqan didn't declare all of its constituent copyrights and licenses)

**Comment by Blahah** [comment]
Putting the whole MIT license in seems horrible. Would a two-liner like this suffice?

```
# Copyright Richard Smith-Unna and Chris Boursnell 2015
# This file is MIT licensed. Full license at: https://github.com/Blahah/transrate/blob/master/LICENSE
```

**Comment by Blahah** [comment]
Because honestly I'd rather just waive all rights in a short one-liner than have a big header in every file! I would feel sick every time I saw it!

**Comment by mr-c** [comment]
:-)

This variation allows both the copyright and license to be detected:

``` shell
mcrusoe@localhost:~$ cat test.py
#!/usr/bin/env python
# Copyright 2015 Richard Smith-Unna and Chris Boursnell.
# This file is MIT licensed.
# Full license at: http://opensource.org/licenses/mit-license.php

import this
mcrusoe@localhost:~$ licensecheck --copyright test.py 
test.py: MIT/X11 (BSD like)
  [Copyright: 2015 Richard Smith-Unna and Chris Boursnell]
```

**Comment by Blahah** [comment]
OK, deal ;).

**Comment by mr-c** [comment]
@Blahah https://wiki.debian.org/Teams/Ruby/RubyExtras/UpstreamDevelopers :-)

**Comment by mr-c** [comment]
Transrate-tools has been accepted into Debian! https://tracker.debian.org/pkg/transrate-tools

**Comment by mr-c** [comment]
snap-aligner has been accepted: https://tracker.debian.org/pkg/snap-aligner

I've received a series of apologies from a debian-ruby team member apologizing for the delays in reviewing my ruby packages.

**Comment by mr-c** [comment]
I'm preparing the salmon package (0.4.1) and I see that version 0.6 is out. @Blahah, when do you expect transrate to be updated to handle the format changes?

**Comment by Blahah** [comment]
@mr-c we'll push a small patch soon addressing reviewer's comments, after which the next release will support the latest salmon. I guess late feb.

Thanks for all your hard work on this :)
